### PR TITLE
bureau has hs reg process TM-3508

### PR DESCRIPTION
**Pull request by SidelineCory24**
To-do
- [x] verify that the overall open bid cycle, submit bid, handshake offer/accepted/registered process still works as expected
- [x] when a cdo registers a handshake on a position that a bureau has not offered a handshake on, the handshake flag shows for the bidder/cdo/bureau on the position and the bureau can not offer handshakes to any of the bidders
- [x] when a cdo unregisters a handshake on a position that a bureau has not offered a handshake on, the handshake does not show for the bidder/cdo/bureau on the position and the bureau can offer handshakes to any of the bidders

**Note: be sure to test with each individual user to avoid role conflict issues**

### CODE REVIEW

**Comment by SidelineCory24** [comment]
I'm getting these dependabot errors for our github yaml file (not sure why the error is happening on my pr lol)
Error 1
image

Error 2
image


Do any of these fixes make since?
image

image


image

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez I did some research and I think these are the dependabot errors my PR is getting for our github yaml file (not sure why the error is happening on my pr lol) **Error 1** image
> 
> **Error 2** image
> 
> Do any of these fixes make since? **Fix 1** image
> 
> **Fix 2** image
> 
> ****Fix 3**** image

Hey Cory! Thanks for looking into this. I think it was showing on your PR just cause of unlucky timing 😅 That yaml pr was recently merged. I opened up this [PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2369) to address this and put you as the reviewer

**Comment by SidelineCory24** [comment]
> > @elizabeth-jimenez I did some research and I think these are the dependabot errors my PR is getting for our github yaml file (not sure why the error is happening on my pr lol) **Error 1** image
> > **Error 2** image
> > Do any of these fixes make since? **Fix 1** image
> > **Fix 2** image
> > ****Fix 3**** image
> 
> Hey Cory! Thanks for looking into this. I think it was showing on your PR just cause of unlucky timing 😅 That yaml pr was recently merged. I opened up this [PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2369) to address this and put you as the reviewer

thanks for this! Test are passing now.

File: src/Components/BureauPage/PositionManagerBidders/PositionManagerBidders.jsx [hunk 1]
```diff
@@ -267,6 +267,7 @@ class PositionManagerBidders extends Component {
     const handshakeRegistered = get(m, 'handshake_registered') === 'Y';
     const active_hs_perdet = get(m, 'active_handshake_perdet');
     const hasAcceptedOtherOffer = get(m, 'has_accepted_other_offer');
+    const hasHsReg = props.hasHsReg;
```
Comment (elizabeth-jimenez) [inline src/Components/BureauPage/PositionManagerBidders/PositionManagerBidders.jsx:270]: Can we change to something like `positionHasHsReg` to differentiate from the one on 267 - i know you didn't name this either, I think it was me 😭

Comment (elizabeth-jimenez) [reply src/Components/BureauPage/PositionManagerBidders/PositionManagerBidders.jsx:270]: oh and a get on the prop value

Comment (SidelineCory24) [reply src/Components/BureauPage/PositionManagerBidders/PositionManagerBidders.jsx]: done

**Comment by elizabeth-jimenez** [comment]
image
Overlay handles following scenario:

When a Bureau HS is already offered to a bidder and the CDO goes and registers it to someone else, the Bureau's choice of bidder can no longer take an action on the Bureau HS

**Review by elizabeth-jimenez — APPROVED** [review]
🧙‍♂️

**Pull request merged**